addons/pos_online_payment_self_order/static/src/overrides/services/pos_store.js

```diff
@@ -37,7 +37,9 @@ patch(PosStore.prototype, {
                 orderId,
             ]);
             const order = result["pos.order"][0];
-            await this.sendOrderInPreparation(order, { bypassPdis: true });
+            if (this.config.self_ordering_pay_after === "each") {
+                await this.sendOrderInPreparation(order, { bypassPdis: true });
```

Comment:
```suggestion
                await this.sendOrderInPreparation(order);
```
Since `bypassPdis` is not used anywhere, we can remove it.